### DISCUSSION TITLE
'this' is casted to another type in the middle of a function

### DISCUSSION THREAD

**Discussion by nbjk667** [NationalSecurityAgency/ghidra#6807] [Q&A]
Good day, I have a question and I am wondering if anyone can help me 🤔

I am decompiling C++ code that essentially looks like this:

```
class BaseClass {
	virtual bool IsDerivedType() { return false; }
	void DoSomething();
};

class DerivedClass : BaseClass {
	virtual bool IsDerivedType() { return true; }
};

void BaseClass::DoSomething() {
	// ...
	if ( IsDerivedType() ) {
		static_cast<DerivedClass*>(this)->...
		// do something with the DerivedClass
	}
}
```

A function is declared in the BaseClass but a portion of the function is using 'this' as if it was DerivedClass.
The problem is that since 'this' is an auto-generated variable I cannot change its type nor it seems to be possible to split it out as a new variable... Is there a workaround for that?

I am new to Ghidra so I am hoping maybe there's some feature I don't know about?

~Robert

**Comment by f-raZ0R** [comment]
Also been encountering a ton of cases like this, and not knowing how to deal with them as ghidra doesn't give me the contextual option to split it out.
bumping this discussion to hopefully get an answer.
